### PR TITLE
Public project gallery: single app view

### PR DESCRIPTION
**Pull request by caleybrock**
Follow up to: https://github.com/code-dot-org/code-dot-org/pull/14869
Spec: https://docs.google.com/document/d/1-TnD9vxq7vzn3ryzCZbGj64MAgF93IwLgz-OGuY0A-Q/edit#

project

Did some re-factoring of our views to be able to create a navigation for a single lab project page. Also changed the default image because it was hard for me to scan for defaults (I'm sure we'll change it later).

Up next:
- load more button on the single app page
- gallery switcher UI
- general UI polish
- what does the default look like if there is no project title or name or image?

### CODE REVIEW

File: apps/src/templates/projects/projectConstants.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  type: PropTypes.string.isRequired,
+  publishedAt: PropTypes.string.isRequired,
+  publishedToPublic: PropTypes.bool.isRequired,
+  publishedToClass: PropTypes.bool.isRequired,
+});
+
+export const projectPropType = PropTypes.shape({
+  projectData: projectDataPropType.isRequired,
+  currentGallery: PropTypes.string.isRequired,
+});
```
Comment (davidsbailey) [inline apps/src/templates/projects/projectConstants.js:18]: Nice extraction!

**Comment by davidsbailey** [comment]
To answer one of your questions, I don't think we'll have to worry too much about projects without thumbnail images or project names -- the script I gave you to run locally to publish a bunch of projects had a shortcoming in that it published "hidden" or channel-backed levels which tend not to have names or thumbnails. In production, I would expect all projects to have a name (even if just "Game Lab Project") and more than half of them to have thumbnail images.

I've just merged https://github.com/code-dot-org/code-dot-org/pull/15026 which allows you to publish and unpublish standalone apps from the My Projects page (but not channel-backed apps, which are hidden in that view). If you'd like to unpublish all your applab and gamelab apps and then manually re-publish the ones you want to see in the gallery, you could run the following from within `bin/pegasus-sql`:
```
update storage_apps set published_at=NULL where project_type = 'applab';
update storage_apps set published_at=NULL where project_type = 'gamelab';
```

**Comment by caleybrock** [comment]
1 apps failure is unrelated so I'm going to merge: 
Unexpected call to console.error: Warning: Failed prop type: Required prop `isPlc` was not specified in `ProgressDetailToggle`.

**Comment by Erin007** [comment]
Little style tweak: do we want the right arrow next to links? 
screen shot 2017-05-12 at 9 47 44 am

**Comment by caleybrock** [comment]
@Erin007 - I'm saving a bunch of style tweaks for another PR because there is still a bit to do to get this to spec. I should have said that in the description, but yes, arrows are to come!

**Pull request merged**